File: bot-api/python/src/robocode_tank_royale/bot_api/mapper/game_setup_mapper.py
from ..game_setup import GameSetup

from robocode_tank_royale.schema import GameSetup as SchemaGameSetup


class GameSetupMapper:
    """Utility class for mapping game setups."""

    @staticmethod
    def map(source: SchemaGameSetup) -> GameSetup:
        """Map a schema GameSetup to a bot-api GameSetup."""

        return GameSetup(
            source.game_type,
            source.arena_width,
            source.arena_height,
            source.number_of_rounds,
            source.gun_cooling_rate,
            source.max_inactivity_turns,
            source.turn_timeout,
            source.ready_timeout,
        )

File: bot-api/typescript/src/mapper/GameSetupMapper.ts
import { GameSetup as SchemaGameSetup } from "../protocol/schema.js";
import { GameSetup } from "../GameSetup.js";

/** Maps schema GameSetup to API GameSetup. */
export class GameSetupMapper {
  static map(s: SchemaGameSetup): GameSetup {
    return new GameSetup(
      s.gameType,
      s.arenaWidth,
      s.arenaHeight,
      s.numberOfRounds,
      s.gunCoolingRate,
      s.maxInactivityTurns,
      s.turnTimeout,
      s.readyTimeout,
    );
  }
}
